zerver/webhooks/statuspage/view.py

```diff
@@ -0,0 +1,40 @@
+# Webhooks for external integrations.
+
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_stream_message
+from zerver.lib.response import json_success
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import get_client, UserProfile
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Text
+
+
+@api_key_only_webhook_view('Statuspage')
+@has_request_variables
+def api_statuspage_webhook(request: HttpRequest, user_profile: UserProfile,
+                           payload: Dict[str, Any]=REQ(argument_type='body'),
+                           stream: str=REQ(default='statuspage-test'),
+                           topic: str=REQ(default='Statuspage')) -> HttpResponse:
+
+    status = payload["page"]["status_description"]
+
+    if status == "All Systems Operational":
+        name = payload["incident"]["name"]
+        state = payload["incident"]["status"]
+        content = payload["incident"]["incident_updates"][0]["body"]
+        body = u"Incident Report :\n **{}** \n * State: **{}** \n * Description: {} \n"
+        body = body.format(name, state, content)
+
+    if status != "All Systems Operational":
+        name = payload["component"]["name"]
+        old_status = payload["component_update"]["old_status"]
+        new_status = payload["component_update"]["new_status"]
+        body = u"Component status changed :\n **{}** has changed status from **{}** to **{}**"
+        body = body.format(name, old_status, new_status)
+
```

Comment:
Okay so:
1. Split out each `if` clause into a separate function. See [how our GCI webhook does this](https://github.com/zulip/zulip/blob/master/zerver/webhooks/gci/view.py#L118).
2. Do not have a space before the `:`. So use `Component status changed:`. :)
3. I think the topic should be the `status_description` and the `name` of the incident, which would mean that you would not need things like `Incident Report:` or `Component status changed:`.